### PR TITLE
Suppress trait errors that are implied by other errors

### PR DESCRIPTION
**Pull request by arielb1**
this is currently a hack and should be cleaned up somewhat. Posting this to get some feedback.

r? @nikomatsakis
cc @estebank

### CODE REVIEW

File: src/test/compile-fail/range_traits-1.rs [hunk 1]
```diff
@@ -17,77 +17,70 @@ use std::ops::*;
 struct AllTheRanges {
     a: Range<usize>,
     //~^ ERROR PartialOrd
```
Comment (estebank) [inline src/test/compile-fail/range_traits-1.rs]: While you're at it, could you clean these up to use `//~|` instead of `//~^^^^^`?

**Review by nikomatsakis — COMMENTED** [review]
The general approach seems reasonably good to me! Sorry for taking so long to give feedback. I'm a bit unsure if we should maybe try to go further?

**Comment by carols10cents** [comment]
Friendly ping @arielb1, keeping this on your radar!

**Comment by arielb1** [comment]
status: now that I'm done with the segfaults on #41917 I'll look into this - I'll try to get it done by Monday,

**Comment by aidanhs** [comment]
Hi @arielb1, how are you getting on with this?

**Comment by arielb1** [comment]
I am. Just a little bit busy with other stuff.

**Comment by nikomatsakis** [comment]
@arielb1 I'm going to close this PR for now just to clear up my queue. Feel free to re-open once you have updates available!

**Comment by Mark-Simulacrum** [comment]
@nikomatsakis You never closed this PR, so I'm going to assume that was unintentional and close it now. Please do reopen if I'm wrong!

**Comment by nikomatsakis** [comment]
@arielb1 I guess you'd prefer to keep the PR open? :) seems fine.

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
@bors r+

(if you feel, maybe address the nit that @estebank pointed out about converting `//~^+` to `//~|`, but doesn't seem criticial to me)

**Comment by arielb1** [comment]
- #42693 (?)
@bors retry

**Pull request merged**